plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveModule.java

```diff
@@ -175,4 +178,16 @@ public boolean translateHiveViews(HiveConfig hiveConfig)
     {
         return hiveConfig.isTranslateHiveViews();
     }
+
+    public record ShutdownExecutors(
```

Comment:
can we have wrapper around Executor, ScheduledExecutor, ExecutorService, ScheduledExecutorService ...
which delegates all the methods and puts ` @PreDestroy` on shutdown?

Would that be nicer?

Reply:
nice

Reply:
that's probably why i couldn't find the Cleanup class (got removed https://github.com/trinodb/trino/pull/20960/commits/0d4f855ecaec81bea194d1fb2a7926f69fff6160#r1514739127)